### PR TITLE
improve support for controllers on mobile, autohide touch controls if present

### PR DESCRIPTION
**Pull request by SquillWall**
This is intended as a fix for the (admittedly small) number of people who use controllers on their mobile devices. This code will perform a check on controller input to automatically hide the touch controls, and if a new touch event is detected, will automatically re-enable them.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -26,6 +26,16 @@ window.addEventListener("unhandledrejection", (event) => {
   //alert(errorString);
 });
 
+//catch all touch events at the top level and perform a check on whether or not the touch controls are already visible.
+//If they are already visible, ignore the event, but otherwise, show the touch controls. This should allow for both gamepad and mobile support
+//without the need for the touch and gamepad settings.
```
Comment (torranx) [inline src/main.ts:31]: nit: Capitalize + spaces after //

**Comment by torranx** [comment]
Codewise looks good but can you possibly provide testing video?

**Comment by SquillWall** [comment]
[mobileControllerSupport.webm](https://github.com/pagefaultgames/pokerogue/assets/170910983/c82b819a-73fc-406e-b42c-a9df00d37930) This is an example of the touch controls being hidden and shown on demand when controller input/touch input is detected

**Comment by SquillWall** [comment]
as a note to testers: General test procedures are as follows...
1. Launch pokerogue on a mobile device
2. Connect an external controller or gamepad to said mobile device
3. Ensure that touch controls are hidden when input is registered from the gamepad
4. Ensure that the touch controls come back when the screen is tapped

**Comment by SquillWall** [comment]
I'm certainly aware the new method of getting the touch controls value in main is a bit of a hack so i'm open to feedback on a better way to pull the value of that setting

**Comment by flx-sta** [comment]
ℹ️ PR considered stale. No actions in the last 30 days.

-----

❗ Feel free to ping me/comment if you want to re-open it

**Pull request closed without merging**